src/main/java/org/mule/module/xml/internal/operation/XPathOperation.java

```diff
@@ -87,7 +87,7 @@ public XPathOperation(EntityExpansion expandEntities) {
                                .evaluate(toDOMNode(content, documentBuilderFactory), contextProperties));
   }
 
-  private Collection<NamespaceMapping> mergeNamespaces(List<NamespaceMapping> namespaces, NamespaceDirectory directory) {
+  Collection<NamespaceMapping> mergeNamespaces(List<NamespaceMapping> namespaces, NamespaceDirectory directory) {
```

Comment:
Make package private to test. Typically I put @VisibleForTesting annotation from guava, but I don't know if your build consumes that annotation to warn if you use that method in prod code.

Reply:
This method was expecting to dedupe common namespaces but because there is no hashcode and equals it doesn't work